engine/lib/friends.php

```diff
@@ -204,6 +204,10 @@ function _elgg_setup_collections_menu() {
  * @access private
  */
 function _elgg_send_friend_notification($event, $type, $object) {
+	if ($object->relationship != 'friend') {
+		return true;
```

Comment:
For events, returning true or null has the same behavior. We should standardize core handlers but we just haven't yet. This is fine for now.